homeassistant/components/xiaomi_aqara/config_flow.py

```diff
@@ -0,0 +1,188 @@
+"""Config flow to configure Xiaomi Aqara."""
+from functools import partial
+import logging
+
+from getmac import get_mac_address
+import voluptuous as vol
+from xiaomi_gateway import XiaomiGatewayDiscovery
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_NAME, CONF_PORT
+from homeassistant.helpers.device_registry import format_mac
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_DISCOVERY_RETRY,
+    CONF_INTERFACE,
+    CONF_KEY,
+    CONF_PROTOCOL,
+    CONF_SID,
+    DOMAIN,
+    ZEROCONF_GATEWAY,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_GATEWAY_NAME = "Xiaomi Aqara Gateway"
+DEFAULT_INTERFACE = "any"
+DEFAULT_DISCOVERY_RETRY = 3
+
+
+GATEWAY_CONFIG = vol.Schema(
+    {vol.Optional(CONF_INTERFACE, default=DEFAULT_INTERFACE): str}
+)
+GATEWAY_SETTINGS = vol.Schema(
+    {
+        vol.Optional(CONF_KEY): vol.All(str, vol.Length(min=16, max=16)),
+        vol.Optional(CONF_DISCOVERY_RETRY, default=DEFAULT_DISCOVERY_RETRY): vol.All(
```

Comment:
Discovery retry should be part of an options flow and not the main flow.

Reply:
Schould we not just remove that options and set a sensible default (I think 3 times is enough)?
I don't feel this is actually really nessesary.

If you want I will make a options flow now I know how to do this, but I don't feel the need for it.

Reply:
Yeah, if we don't need the option please remove it. That's even better.

Reply:
I have removed that option and set the defaut to 5 retries